### PR TITLE
Added subtree from the azure functions 'protobuf'

### PR DESCRIPTION
**Pull request by daxian-dbw**
1. Added subtree from https://github.com/azure/azure-functions-language-worker-protobuf. Branch: dev. Commit: 58b3dc0
1. Generate the CSharp code for `protobuf` automatically, so we don't need to manually generate the CSharp files when the protocol is changed.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -46,14 +46,19 @@ if($Clean.IsPresent) {
     Pop-Location
 }
 
-# Build step
+# Common step required by both build and test
 Find-Dotnet
+Restore-NuGetPackage
```
Comment (tylerl0706) [inline build.ps1]: Why do we need this? When you do a `dotnet publish` on the sln file in the root, it will `restore` all of the containing projects.

**Comment by tylerl0706** [comment]
Is it normal for the submodule contents to show up in the PR?

File: tools/helper.psm1 [hunk 2]
```diff
@@ -81,6 +82,92 @@ function Install-Dotnet {
...
+        if ($LASTEXITCODE -ne 0) {
+            throw "Failed to generate the CSharp code for gRPC communication."
+        }
+    }
+}
+
+function Resolve-ProtoBufToolPath
+{
+    if (-not $Script:protoc_Path) {
+        Write-Log "Resolve the protobuf tools for auto-generating code"
```
Comment (tylerl0706) [inline tools/helper.psm1]: it can probably be addressed in another PR but is there a reason we have `Write-Log`? Couldn't we just use `Write-Host` and `Write-Warning` etc?

Comment (daxian-dbw) [reply tools/helper.psm1]: I don't want to specify `-ForegroundColor Green` for all the intentional progress logging. `Write-Log` abstracts that flag.
